### ISSUE TITLE
Using HttpMediatR without logger interface

### ISSUE THREAD

**Issue by wimvanhouts** [RPM1984/HttpMediatR#1]
Hi, really like the component, really usefull and exactly what I was looking for. Would you consider making a version available that also works without Microsoft.Exentensions.Logging?

**Comment by RPM1984** [comment]
Hi there, glad you like it and find it useful!

Just curious - why don't you want the logging dependency? Is it because you don't want logging at all, and therefore don't want to pass the logger to the ctor? 

Reluctant to remove logging support altogether, as some consumers (me included) might want some logging info about what's going on under the covers.

Maybe a nice compromise might be a ctor overload that doesn't require the `ILogger`.. would that work for you?

Let me know your thoughts. 🙂

**Comment by wimvanhouts** [comment]
Ctor overload without ILogger requirement would be perfect. Reason for not using logging extensions: at this point, don't like the logging extensions interface... misses a few critical things I believe are usefull, and, are hard to implement as extensions (still trying to figure that one out when I would have the time). So at this point I don't use it yet, might be in te futuren, but now I always have to provide it. If you provide ctor without param and handle internally, then everybody is happy I guess

**Comment by RPM1984** [comment]
Yep, let's do the optional `ILogger`, that way everyone wins as you say.

Feel free to submit a PR, otherwise I'll try and get that done this week and push up a new NuGet release.

**Comment by wimvanhouts** [comment]
I'm not that experienced in Github OSS development... but sure do want to give it a try. Unfortunatly, code fetched from Github without modifications already fails to build. error CS1705 pops up on Tests project.

Maybe you're more experienced than I am but from what I understand this has to do with the fact that in the samples you reference Microsoft.AspNetCore.App 2.1.1 while in the test still reference  2.1.0. Changed that in the project of the tests and this fixed the error. Will make that part of the PR as well

> 3>CSC : error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(31,30,31,94): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(32,88,32,94): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(32,81,32,94): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(32,31,32,80): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.Extensions.Logging.Abstractions' with identity 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(56,30,56,94): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(57,88,57,94): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(57,81,57,94): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(57,31,57,80): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.Extensions.Logging.Abstractions' with identity 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(74,30,74,96): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(75,90,75,96): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(75,83,75,96): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(75,31,75,82): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.Extensions.Logging.Abstractions' with identity 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(102,30,102,96): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(103,90,103,96): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(103,83,103,96): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(103,31,103,82): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.Extensions.Logging.Abstractions' with identity 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(118,30,118,96): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(119,90,119,96): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(119,83,119,96): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(119,31,119,82): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.Extensions.Logging.Abstractions' with identity 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(134,30,134,97): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(135,91,135,97): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(135,84,135,97): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(135,31,135,83): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.Extensions.Logging.Abstractions' with identity 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(150,30,150,97): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(151,91,151,97): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(151,84,151,97): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.AspNetCore.Mvc.Abstractions' with identity 'Microsoft.AspNetCore.Mvc.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>HttpHandlerTests.cs(151,31,151,83): error CS1705: Assembly 'HttpMediatR.Samples.AspNetCoreMvc' with identity 'HttpMediatR.Samples.AspNetCoreMvc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' which has a higher version than referenced assembly 'Microsoft.Extensions.Logging.Abstractions' with identity 'Microsoft.Extensions.Logging.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
> 3>Done building project "HttpMediatR.Tests.csproj" -- FAILED.

**Comment by RPM1984** [comment]
Weird...I remember having that error originally when upgrading to .net core 2.1, but thought I fixed it (proof is CI is working)

I'll check it out again, maybe on a fresh cloned git repo..

But in the meantime, yep just include in your PR.

Welcome to Github OSS! 😀 (I'm not that experienced either, so we're both learning)

**Comment by wimvanhouts** [comment]
Hi, I get a 403 when trying to push my branch (in preparation of PR). Is it sure this should be possible that I push branches/create PR's on your project? Could be someting on my machine, but, git config shows correct email at least... don't know if that is sufficient

**Comment by RPM1984** [comment]
So you can't push directly (eg 'write') to this repository.

The way most OSS projects work is the fork + pull request style.

See here: https://guides.github.com/activities/forking/

**Comment by wimvanhouts** [comment]
Ok, will do it as such. I'm however biased to follow the reasoning in this document https://blog.gitprime.com/the-definitive-guide-to-forks-and-branches-in-git/, which, would suggest branching would be te appropriate way to do this feature implementation. Will fork this evening and perform a PR as such

**Comment by RPM1984** [comment]
Interesting article, and raises some good points.

My personal opinion however is that branching makes sense for internal projects, not public ones where the owner wants to control what code goes into the main code base.

This way, only approved/reviewed code makes it's way into the origin.

Just my 2c 😊

**Comment by wimvanhouts** [comment]
Agreed, though, the same is possible with branches if you prevent people to merge into your master. But ok, valid points for both ways, so will go your way, no problem.

**Comment by RPM1984** [comment]
Yeah, basically I'd rather have write protection at the _repo_ level, instead of _branch_ level, for the following reasons: 
- don't need to update my CI to only build on master
- don't need special branch protection in GitHub
- don't need to worry about people creating a bunch of branches and increasing the repository size, and needing to clean them up if they aren't merged etc

Like I said, I'm pretty new to OSS too..so my no means is my way right. 😀

Look forward to the PR 🎉

**Comment by RPM1984** [comment]
Fixed by [PR #2](https://github.com/RPM1984/HttpMediatR/pull/2)

**Comment by RPM1984** [comment]
Damn... CI is failing now with this:

```
2018-07-04T04:51:58.2398260Z Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: Testhost process exited with error: It was not possible to find any compatible framework version
2018-07-04T04:51:58.2398802Z The specified framework 'Microsoft.AspNetCore.App', version '2.1.1' was not found.
2018-07-04T04:51:58.2399132Z   - Check application dependencies and target a framework version installed at:
2018-07-04T04:51:58.2399378Z       C:\Program Files\dotnet\
2018-07-04T04:51:58.2399627Z   - Installing .NET Core prerequisites might help resolve this problem:
2018-07-04T04:51:58.2399869Z       http://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409
2018-07-04T04:51:58.2400323Z   - The .NET Core framework and SDK can be installed from:
2018-07-04T04:51:58.2400529Z       https://aka.ms/dotnet-download
2018-07-04T04:51:58.2400747Z   - The following versions are installed:
2018-07-04T04:51:58.2401010Z       2.1.0 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
2018-07-04T04:51:58.2401191Z 
2018-07-04T04:51:58.2401452Z    at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources, CancellationToken cancellationToken)
2018-07-04T04:51:58.2401847Z    at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyExecutionManager.StartTestRun(TestRunCriteria testRunCriteria, ITestRunEventsHandler eventHandler)
2018-07-04T04:51:58.3009908Z 
2018-07-04T04:51:58.3011967Z Test Run Aborted.
2018-07-04T04:51:58.3696082Z ##[error]Error: C:\Program Files\dotnet\dotnet.exe failed with return code: 1
2018-07-04T04:51:58.3710928Z ##[warning]No test result files were found.
2018-07-04T04:51:58.3712377Z ##[error]Dotnet command failed with non-zero exit code on the following projects : D:\a\1\s\tests\HttpMediatR.Tests\HttpMediatR.Tests.csproj
```

So guessing we need to update some other package? Did the tests work when running locally?

I didn't have the VSTS CI setup properly for PR's, so i couldn't validate the build checks with your PR.

So i'll need to fix this before i push a new NuGet release.. (sorry, been super busy!)

**Comment by wimvanhouts** [comment]
Tests and all build on my machine perfectly. Are you sure your CI build machine is up to date with the latest versions? It seems to me the 2.1.1 framework is not installed and since the test are now on that version, it is  a requirment for running them. I did however notice strange behaviour in other projects where the build worked fine for a project in 2.1.1 and the run failed. This was solved by installing asp net core version 2.1.301 ... while with 2.1.300 it failed.

**Comment by RPM1984** [comment]
@wimvanhouts i _finally_ got a quick chance to check this out.

I got a weird error on my machine trying to get the tests to run. Seems like there's a weird runtime mismatch.

For now, i downgraded back to .NET Core 2.1.0, and everything is fine now.

I've published a new release to NuGet with your work. Enjoy! 😎
